armory/scenarios/ucf101.py

```diff
@@ -0,0 +1,197 @@
+                ):
+                    x_trains, y_trains = train_data_generator.get_batch()
+                    # x_trains consists of one or more videos, each represented as an
+                    # ndarray of shape (n_stacks, 3, 16, 112, 112).
+                    # To train, randomly sample a batch of stacks
+                    x_train = np.zeros(
+                        (min(batch_size, len(x_trains)), 3, 16, 112, 112),
+                        dtype=np.float32,
+                    )
+                    for i, xt in enumerate(x_trains):
+                        rand_stack = np.random.randint(0, xt.shape[0])
+                        x_train[i, ...] = xt[rand_stack, ...]
+                    classifier.fit(
+                        x_train, y_trains, batch_size=batch_size, nb_epochs=1
+                    )
+                logger.info(f"Epoch {epoch}/{train_epochs} took {time.time() - start}s")
+
+                # evaluate on test examples
+                classifier.set_learning_phase(False)
+                test_data_generator = load_dataset(
+                    config["dataset"],
+                    epochs=1,
+                    split_type="test",
+                    preprocessing_fn=preprocessing_fn,
+                )
+
+                accuracies = AverageMeter()
+                accuracies_top5 = AverageMeter()
+                video_count = 0
+                for _ in tqdm(range(len(test_data_generator) // 10), desc="Validation"):
+                    x_tests, y_tests = test_data_generator.get_batch()
+                    for x_test, y_test in zip(x_tests, y_tests):
+                        y = classifier.predict(x_test)
+                        y = np.argsort(np.mean(y, axis=0))[-5:][::-1]
+                        acc = float(y[0] == y_test)
+                        acc_top5 = float(y_test in y)
+                        accuracies.update(acc, 1)
+                        accuracies_top5.update(acc_top5, 1)
+                logger.info(
+                    f"Top-1 video accuracy = {accuracies.avg}, "
+                    f"top-5 video accuracy = {accuracies_top5.avg}"
+                )
+
+        # Evaluate ART classifier on test examples
+        logger.info("Running inference on benign test examples...")
+        logger.info(f"Loading testing dataset {config['dataset']['name']}...")
+        classifier.set_learning_phase(False)
+        test_data_generator = load_dataset(
+            config["dataset"],
+            epochs=1,
+            split_type="test",
+            preprocessing_fn=preprocessing_fn,
+        )
+
+        test_accuracies = AverageMeter()
+        test_accuracies_top5 = AverageMeter()
+        video_count = 0
+        for x_tests, y_tests in tqdm(test_data_generator, desc="Benign"):
+            for x_test, y_test in zip(x_tests, y_tests):
+                y = classifier.predict(x_test)
+                y = np.argsort(np.mean(y, axis=0))[-5:][::-1]
```

Comment:
Unclear of where `[-5:]` is coming from?

Reply:
It's calculating top_5 accuracy - I'd like to put this into the `armory.util.metrics` file and call it, though that may want to wait for the ImageClassification unification.